### PR TITLE
[ONNX] Create common ExportTestCase base class

### PR DESCRIPTION
**Pull request by BowenBao**
Refactor out a common base class `ExportTestCase`, for common things in `setUp`.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88261
* #87258
* __->__ #88145
* #87830

### CODE REVIEW

File: test/onnx/pytorch_test_common.py [hunk 1]
```diff
@@ -188,3 +193,18 @@ def wrapper(self, *args, **kwargs):
...
     return tuple(function._iter_filter(lambda o: isinstance(o, torch.Tensor))(x))
+
+
+def set_rng_seed(seed):
+    torch.manual_seed(seed)
+    random.seed(seed)
+    np.random.seed(seed)
+
+
+class ExportTestCase(common_utils.TestCase):
```
Comment (justinchuby) [inline test/onnx/pytorch_test_common.py:204]: A brief on when it should be used would be helpful!

File: test/onnx/test_utility_funs.py [hunk 2]
```diff
@@ -27,12 +28,7 @@
 from verify import verify
 
 
-class _BaseTestCase(common_utils.TestCase):
-    def setUp(self):
```
Comment (justinchuby) [inline test/onnx/test_utility_funs.py:31]: Nice

File: test/onnx/pytorch_test_common.py (hunk 1 above)
Comment (justinchuby) [reply test/onnx/pytorch_test_common.py:204]: common_utils.TestCase.setUp has some seed setting behaviors iirc. Can we rely on that for torch and np etc.?

**Review by justinchuby — APPROVED** [review]
LGTM🚀

File: test/onnx/pytorch_test_common.py (hunk 1 above)
Comment (BowenBao) [reply test/onnx/pytorch_test_common.py:204]: Good idea, uncovered several flaky tests. Reverting and tracking through https://github.com/pytorch/pytorch/issues/88264

**Review by abock — APPROVED** [review]

File: test/onnx/pytorch_test_common.py (hunk 1 above)
Comment (AllenTiTaiWang) [reply test/onnx/pytorch_test_common.py:204]: I wonder if we could extend this for tests other than runtime cases

**Review by titaiwangms — APPROVED** [review]
🏆

**Pull request closed without merging**